controllers/apply/forms/create-form-model.js

```diff
@@ -0,0 +1,113 @@
+const { find, flatMap, has, get } = require('lodash');
+
+/**
+ * For a given field attach some additional computed properties
+ * - isConditionalOr flags if this field is a conditional field, i.e. has a dependency on another field (child field).
+ * - isConditionalFor flags if a field could trigger any conditional fields (parent field).
+ * - conditionalFor attaches information about any conditional fields associated with a parent field (names of the conditional fields and the value to check for)
+ */
+function enhanceField(field, associatedConditionalFields) {
+    const conditionalFor = associatedConditionalFields.map(_ => {
+        return {
+            triggerField: _.name,
+            triggerOnValue: _.conditionalOn.value
+        };
+    });
+
+    return Object.assign(field, {
+        isConditionalOn: has(field, 'conditionalOn'),
+        isConditionalFor: conditionalFor.length > 0,
+        conditionalFor: conditionalFor
+    });
+}
+
+function getFieldsForFieldsets(fieldsets) {
+    return function() {
+        const allFields = flatMap(fieldsets, fieldset => fieldset.fields);
+        const conditionalFields = allFields.filter(field => has(field, 'conditionalOn'));
+        const enhancedFields = allFields.map(field => {
+            const associatedConditionalFields = conditionalFields.filter(_ => {
+                const conditionalOn = get(_, 'conditionalOn', {});
+                return field.name === conditionalOn.name;
+            });
+
+            const enhancedField = enhanceField(field, associatedConditionalFields);
+            return enhancedField;
+        });
+
+        return enhancedFields;
+    };
+}
+
+/**
+ * Create a step based on a schema.
+ * Allows us to pass a relatively consise schema for the step,
+ * this funciton then adds some additional computed methods on top.
+ * - withValues allows the current form data for a step to be passed in and the values attached to each field
+ * - getValidators collects all validators associated with each field for express-valdiator
```

Comment:
Typo on "valdiator" (although you're in [good company](https://github.com/search?o=desc&q=valdiator&s=indexed&type=Code&utf8=%E2%9C%93)).